cmd/upgrade/accountroles/cmd.go

```diff
@@ -311,66 +279,25 @@ func buildCommands(prefix string, accountID string, isUpgradeNeedForAccountRoleP
 		for file, role := range aws.AccountRoles {
 			name := aws.GetRoleName(prefix, role.Name)
 			policyARN := aws.GetPolicyARN(accountID, fmt.Sprintf("%s-Policy", name), policyPath)
-			iamRoleTags := map[string]string{
-				tags.OpenShiftVersion: defaultPolicyVersion,
-			}
-
-			tagRole := awscb.NewIAMCommandBuilder().
-				SetCommand(awscb.TagRole).
-				AddTags(iamRoleTags).
-				AddParam(awscb.RoleName, name).
-				Build()
-			//check if the policy exists if not output create and attach
 			_, err := awsClient.IsPolicyExists(policyARN)
-			if err != nil {
-				policyName := fmt.Sprintf("%s-Policy", name)
-				iamTags := map[string]string{
-					tags.OpenShiftVersion: defaultPolicyVersion,
-					tags.RolePrefix:       prefix,
-					tags.RoleType:         file,
-					tags.RedHatManaged:    "true",
-				}
-
-				createPolicy := awscb.NewIAMCommandBuilder().
-					SetCommand(awscb.CreatePolicy).
-					AddParam(awscb.PolicyName, policyName).
-					AddParam(awscb.PolicyDocument, fmt.Sprintf("file://sts_%s_permission_policy.json", file)).
-					AddTags(iamTags).
-					AddParam(awscb.Path, policyPath).
-					Build()
-
-				attachRolePolicy := awscb.NewIAMCommandBuilder().
-					SetCommand(awscb.AttachRolePolicy).
-					AddParam(awscb.RoleName, name).
-					AddParam(awscb.PolicyArn, aws.GetPolicyARN(accountID, policyName, policyPath)).
-					Build()
-
-				_, _ = awsClient.IsRolePolicyExists(name, policyName)
-				if err != nil {
-					commands = append(commands, createPolicy, attachRolePolicy, tagRole)
-				} else {
-					deletePolicy := awscb.NewIAMCommandBuilder().
-						SetCommand(awscb.DeleteRolePolicy).
-						AddParam(awscb.RoleName, name).
-						AddParam(awscb.PolicyName, policyName).
-						Build()
-					commands = append(commands, createPolicy, attachRolePolicy, tagRole, deletePolicy)
-				}
-			} else {
-				createPolicyVersion := awscb.NewIAMCommandBuilder().
-					SetCommand(awscb.CreatePolicyVersion).
-					AddParam(awscb.PolicyArn, policyARN).
-					AddParam(awscb.PolicyDocument, fmt.Sprintf("file://sts_%s_permission_policy.json", file)).
-					AddParamNoValue(awscb.SetAsDefault).
-					Build()
-
-				tagPolicies := awscb.NewIAMCommandBuilder().
-					SetCommand(awscb.TagPolicy).
-					AddTags(iamRoleTags).
-					AddParam(awscb.PolicyArn, policyARN).
-					Build()
-				commands = append(commands, createPolicyVersion, tagPolicies, tagRole)
-			}
+			hasPolicy := err == nil
+			policyName := fmt.Sprintf("%s-Policy", name)
+			_, err = awsClient.IsRolePolicyExists(name, policyName)
+			hasInlinePolicy := err == nil
```

Comment:
Ideally, we would return an additional boolean value from `IsRolePolicyExists` as the name implies.
But this is out of the scope of this MR, so it's up to you.

Reply:
Yes, I agree. It did felt weird